sklearn/utils/__init__.py

```diff
@@ -1056,7 +1056,7 @@ def _approximate_mode(class_counts, n_draws, rng):
     rng = check_random_state(rng)
     # this computes a bad approximation to the mode of the
     # multivariate hypergeometric given by class_counts and n_draws
-    continuous = n_draws * class_counts / class_counts.sum()
+    continuous = class_counts / class_counts.sum() * n_draws
```

Comment:
Also please add a test for overflow in this function (e.g. the example you put in the issue).

Reply:
Done